### PR TITLE
fix: Do not ask for external write permissions on android 13 and above

### PR DESCRIPTION
**Pull request by gongracr**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

On devices running Android 13 (Tiramisu) and above, Google reworked the permissions requests logic so that it will not be needed to request the `WRITE_EXTERNAL_STORAGE` anymore. If this permission is still checked from that version onwards, it will always return `false`.

### Solutions
Remove the request for permissions to write in external storage as it is not needed when using a content resolver. That way we will unify the way to save files externally in both the gallery and the conversation view models. Also removed the permission from Manifest.

#### How to Test

Have a device with Android 13 or above. Receive an asset. Try to save it and check it always succeeds

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1263/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5284363)(5284363)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5284363)(5284363)-dev-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/DownloadedAssetDialog.kt [hunk 1]
```diff
@@ -28,19 +25,6 @@ fun DownloadedAssetDialog(
         val assetSize = downloadedAssetDialogState.assetSize
         val messageId = downloadedAssetDialogState.messageId
 
-        // Flow to get write to external storage permission
-        val requestWriteToExternalStorageRequestFlow = WriteToExternalStorageRequestFlow(
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/DownloadedAssetDialog.kt:32]: will removing this brake is for Android < 13 ?

Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/DownloadedAssetDialog.kt:32]: Nope. It turned out we don't need to request write permissions when writing AR internal files to download folder when using the content provider also for Android < 13. Actually this is how we were already doing it in the GalleryViewModel.
I tested it on Android 12 and 13 and everything seems fine.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1263/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5307594)(5307594)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5307594)(5307594)-dev-compat.apk)

**Review by tmspzz — APPROVED** [review]

**Pull request merged**